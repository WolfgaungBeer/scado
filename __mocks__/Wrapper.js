import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { Theme, defaultTheme } from '../scado';

const mockStore = configureStore();
const store = mockStore({
    scado: {
        theme: defaultTheme,
        toasts: [
            { id: '1', type: 'success', content: 'Text', onClose: () => {} },
            { id: '2', type: 'error', content: 'Text', onClose: () => {} },
        ],
    },
});


const Wrapper = ({ children }) => ( // eslint-disable-line
    <Provider store={store}>
        <Theme>{children}</Theme>
    </Provider>
);

export default Wrapper;
