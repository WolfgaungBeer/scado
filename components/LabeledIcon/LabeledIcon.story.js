import React from 'react';
import { storiesOf } from '@storybook/react';
import { text, boolean } from '@storybook/addon-knobs';
import LabeledIcon from './';


storiesOf('LabeledIcon', module)
    .add('default', () => (
        <LabeledIcon icon="dashboard" label="Ein Label" />
    ))
    .add('with props', () => (
        <LabeledIcon
            icon={text('icon', 'dashboard')}
            label={text('label', 'Ein Label')}
            color={text('color', 'success')}
            scale={text('scale', 'l')}
            loading={boolean('loading', false)}
            invert={boolean('invert', false)}
        />
    ));
