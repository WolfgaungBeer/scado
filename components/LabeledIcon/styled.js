import styled from 'styled-components';

export const StyledLabeledIcon = styled.div`
    position: relative;
    padding-top: 0.25em;
    color: ${props => props.theme.color[props.color]};
    font-size: ${props => props.theme.scale[props.scale]};
`;

export const StyledText = styled.span`
    position: relative;
    ${props => (props.invert ? 'right: 0.2rem' : 'left: 0.2rem')};
    padding: 0 0.25rem;
    bottom: 0.15em;
`;
