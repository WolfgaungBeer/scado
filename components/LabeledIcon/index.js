import React from 'react';
import { string, node, bool } from 'prop-types';
import { StyledLabeledIcon, StyledText } from './styled';
import Icon from '../Icon';


const propTypes = {
    icon: string.isRequired,
    label: node.isRequired,
    color: string,
    scale: string,
    loading: bool,
    invert: bool,
};

const defaultProps = {
    color: undefined,
    scale: undefined,
    loading: undefined,
    invert: undefined,
};

const LabeledIcon = ({ icon, label, color, scale, loading, invert }) => {
    const left = invert ? <StyledText invert={invert}>{label}</StyledText> : <Icon icon={icon} loading={loading} />;
    const right = invert ? <Icon icon={icon} loading={loading} /> : <StyledText invert={invert}>{label}</StyledText>;
    return (
        <StyledLabeledIcon color={color} scale={scale}>
            {left}
            {right}
        </StyledLabeledIcon>
    );
};

LabeledIcon.propTypes = propTypes;
LabeledIcon.defaultProps = defaultProps;

export default LabeledIcon;
