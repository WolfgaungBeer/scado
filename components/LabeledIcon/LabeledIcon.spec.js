import React from 'react';
import renderer from 'react-test-renderer';
import Wrapper from '../../__mocks__/Wrapper';
import LabeledIcon from './';

it('renders the Icon Component correctly', () => {
    const tree1 = renderer.create(
        <Wrapper>
            <LabeledIcon icon="dashboard" label="Test" />
        </Wrapper>,
    ).toJSON();
    expect(tree1).toMatchSnapshot();

    const tree2 = renderer.create(
        <Wrapper>
            <LabeledIcon icon="dashboard" label="Test" color="error" scale="l" loading />
        </Wrapper>,
    ).toJSON();
    expect(tree2).toMatchSnapshot();

    const tree3 = renderer.create(
        <Wrapper>
            <LabeledIcon icon="dashboard" label="Test" color="error" scale="l" invert />
        </Wrapper>,
    ).toJSON();
    expect(tree3).toMatchSnapshot();
});
