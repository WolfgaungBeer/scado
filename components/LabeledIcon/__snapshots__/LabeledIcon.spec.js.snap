// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Icon Component correctly 1`] = `
<div
  className="sc-bdVaJa jQGYJy"
  color={undefined}
  scale={undefined}
>
  <i
    className="material-icons  sc-htpNat VYLmK"
    color={undefined}
    scale={undefined}
  >
    dashboard
  </i>
  <span
    className="sc-bwzfXH hrXnsj"
  >
    Test
  </span>
</div>
`;

exports[`renders the Icon Component correctly 2`] = `
<div
  className="sc-bdVaJa iHYzGI"
  color="error"
  scale="l"
>
  <i
    className="material-icons  sc-htpNat eqYdRJ"
    color={undefined}
    scale={undefined}
  >
    autorenew
  </i>
  <span
    className="sc-bwzfXH hrXnsj"
  >
    Test
  </span>
</div>
`;

exports[`renders the Icon Component correctly 3`] = `
<div
  className="sc-bdVaJa iHYzGI"
  color="error"
  scale="l"
>
  <span
    className="sc-bwzfXH kndBaX"
  >
    Test
  </span>
  <i
    className="material-icons  sc-htpNat VYLmK"
    color={undefined}
    scale={undefined}
  >
    dashboard
  </i>
</div>
`;
